package.json

```diff
@@ -2,7 +2,9 @@
   "name": "react-use-event-hook",
   "version": "0.9.2",
   "description": "Same as React's `useCallback`, but returns a stable reference.",
-  "main": "dist/useEvent.js",
+  "main": "dist/index.js",
+  "types": "dist/useEvent.d.ts",
+  "module": "dist/useevent.mjs",
```

Comment:
In general, this sounds like a great addition.  Supporting ESM is a good idea.

However, looking at the compiled `tsup` output, I'm disappointed by the extra complexity.  I'm a fan of simply using `tsc`, and its ESM output is much cleaner and easier to read or debug.  

I would love if you could update this PR to simply use `tsc`, instead of introducing new dependencies.

Reply:
Since I was experimenting with this, I created #10